mmcv/cnn/rfsearch/search.py

```diff
@@ -0,0 +1,250 @@
+# Copyright (c) OpenMMLab. All rights reserved.
+import logging
+import os
+from typing import Dict, Optional
+
+import torch  # noqa
+import torch.nn as nn
+
+import mmcv
+from mmcv.cnn.rfsearch.utils import get_single_padding, write_to_json
+from mmcv.runner import HOOKS, Hook
+from .operator import Conv2dRFSearchOp, ConvRFSearchOp  # noqa
+
+logging.basicConfig(
+    format='[%(asctime)s-%(filename)s-%(levelname)s:%(message)s]',
+    level=logging.NOTSET,
+    datefmt='%Y-%m-%d%I:%M:%S %p',
+)
+logger = logging.getLogger('Searcher')
+logger.setLevel(logging.ERROR)
+
+
+@HOOKS.register_module()
+class RFSearchHook(Hook):
+    """Rcecptive field search via dilation rates.
+        Paper: RF-Next: Efficient Receptive Field
+            Search for Convolutional Neural Networks, TPAMI2022 (CVPR2021)
+            https://arxiv.org/abs/2206.06637
+
+    Args:
+        mode (str, optional): It can be set to the following types:
+            'search', 'fixed_single_branch', or 'fixed_multi_branch'.
+        config (Dict, optional): config dict of search.
+        rfstructure_file (str, optional):
+            Searched receptive fields of the model.
+        by_epoch (bool, optional):
+            Determine perform step by epoch or by iteration.
+            If set to True, it will step by epoch. Otherwise, by iteration.
+            Default: True.
+        verbose (bool):
+            Determines whether to print rf-next related logging messages.
+            Defaults to True.
```

Comment:
```suggestion
        rfstructure_file (str, optional): Path to load searched receptive fields
            of the model. Defaults to None.
        by_epoch (bool, optional): Determine to perform step by epoch or
            by iteration. If set to True, it will step by epoch. Otherwise, by
            iteration. Defaults to True.
        verbose (bool): Determines whether to print rf-next related logging
            messages. Defaults to True.
```